nav2_behavior_tree/include/nav2_behavior_tree/bt_action_node.hpp

```diff
@@ -272,18 +304,44 @@ class BtActionNode : public BT::ActionNodeBase
         }
       };
 
-    auto future_goal_handle = action_client_->async_send_goal(goal_, send_goal_options);
+    future_goal_handle_ = action_client_->async_send_goal(goal_, send_goal_options);
+    sent_time_ = node_->now();
+    goal_sent_ = true;
+  }
 
-    if (rclcpp::spin_until_future_complete(node_, future_goal_handle, server_timeout_) !=
-      rclcpp::FutureReturnCode::SUCCESS)
-    {
+  /**
+   * @brief Function to check response from the action server when a new goal is sent
+   * @return boolen True if future_goal_handle_ returns SUCCESS, False when it returns TIMEOUT
+   */
+  bool check_future_goal_handle()
```

Comment:
Rename maybe to `if_goal_received`, I think semantically it helps explain what's happening here more and what a true vs a false return would actually mean. Maybe not exactly that name, but something similar?